### PR TITLE
Up-132-login-is-unauthenticated

### PR DESCRIPTION
**Pull request by d-username**
#132

### CODE REVIEW

**Review by dogezen — DISMISSED** [review]

File: src/controllers/auth.js [hunk 1]
```diff
@@ -7,9 +7,9 @@ import { sendDataResponse, sendMessageResponse } from '../utils/responses.js'
 export const login = async (req, res) => {
   const { email, password } = req.body
 
-  if (!email) {
-    return sendDataResponse(res, 400, {
-      email: 'Invalid email and/or password provided'
+  if (!email || !password) {
+    return sendDataResponse(res, 401, {
+      email: 'Invalid email or/and password provided'
```
Comment (Auenc) [inline src/controllers/auth.js:12]: We should keep the error messages as they were before `Invalid email and/or password provided`. The reason for this is we don't want potential attackers to know if they've got the correct email address but wrong password.

File: src/controllers/auth.js [hunk 2]
```diff
@@ -18,8 +18,8 @@ export const login = async (req, res) => {
     const areCredentialsValid = await validateCredentials(password, foundUser)
 
     if (!areCredentialsValid) {
-      return sendDataResponse(res, 400, {
-        email: 'Invalid email and/or password provided'
+      return sendDataResponse(res, 401, {
+        email: 'Invalid password provided'
```
Comment (Auenc) [inline src/controllers/auth.js:22]: Same as above

**Review by Auenc — CHANGES_REQUESTED** [review]
Just a small change to the error message then this will be good to go!

**Review by Auenc — APPROVED** [review]
LGTM

**Review by Alex90Jennings — APPROVED** [review]

**Pull request merged**